athena-hbase/src/test/java/com/amazonaws/athena/connectors/hbase/HbaseMetadataHandlerTest.java

```diff
@@ -223,19 +224,54 @@ public void doGetTable()
                 .build();
 
         assertEquals(expectedSchema.getFields().size(), res.getSchema().getFields().size());
-        logger.info("doGetTable - exit");
     }
 
     @Test
-    public void doGetTableLayout()
+    public void doGetTableFromGlue()
             throws Exception
     {
-        logger.info("doGetTableLayout - enter");
+        List<Column> columns = new ArrayList<>();
+        columns.add(new Column().withName("col1").withType("int"));
+        columns.add(new Column().withName("col2").withType("date"));
+        columns.add(new Column().withName("col3").withType("timestamp"));
+
+        Map<String, String> param = ImmutableMap.of(
+                SOURCE_TABLE_PROPERTY, TEST_TABLE,
+                HbaseMetadataHandler.HBASE_METADATA_FLAG, "true",
+                COLUMN_NAME_MAPPING_PROPERTY, "col1=Col1 , col2=Col2 ,col3=Col3",
+                DATETIME_FORMAT_MAPPING_PROPERTY, "col1=datetime1,col3=datetime3 ");
+        com.amazonaws.services.glue.model.Table table = new com.amazonaws.services.glue.model.Table()
+                .withParameters(param)
+                .withPartitionKeys()
+                .withStorageDescriptor(new StorageDescriptor().withColumns(columns));
+        GetTableResult mockResult = new GetTableResult().withTable(table);
+        when(awsGlue.getTable(any())).thenReturn(mockResult);
+
+        GetTableRequest getTableRequest = new GetTableRequest(IDENTITY, QUERY_ID, DEFAULT_CATALOG, TABLE_NAME);
+        GetTableResponse getTableResponse = handler.doGetTable(allocator, getTableRequest);
+        logger.info("doGetTableFromGlue: GetTableResponse[{}]", getTableResponse);
+
+        // 4 including HbaseSchemaUtils.ROW_COLUMN_NAME
+        assertEquals(4, getTableResponse.getSchema().getFields().size());
+
+        Field col1 = getTableResponse.getSchema().findField("Col1");
+        assertEquals(Types.MinorType.INT, Types.getMinorTypeForArrowType(col1.getType()));
+
+        Field col2 = getTableResponse.getSchema().findField("Col2");
+        assertEquals(Types.MinorType.DATEDAY, Types.getMinorTypeForArrowType(col2.getType()));
+
+        Field col3 = getTableResponse.getSchema().findField("Col3");
+        assertEquals(Types.MinorType.DATEMILLI, Types.getMinorTypeForArrowType(col3.getType()));
```

Comment:
I'm confused how we are supporting this from HBase, how do we go from HBase data format to dateday or datemilli? Specifically HbaseSchemaUtils.coerceType(...) has no code for handling Dateday and datemilli so its not possible to query these types. My suggestion is to rollback changes to any connector other than DDB unless you have seen these other connectors successfully query these types from their real source (e.g. not a unit test). Or is this just a new unit test which happened to have these types?

Reply:
removed from metadata Handler. we are not supporting them in record handler for hbase and redis